GoogleUtilities/Logger/Private/GULOSLogger.m

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "GULOSLogger.h"
+
+#import <os/log.h>
+
+#import "GULAppEnvironmentUtil.h"
+#import "GULLogger+Internal.h"
+#import "GULLoggerLevel.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+// Function which calls the macro so that this can be substituted for testing.
+// Since the macro enforces built-in constant-ness of the format string, it is replaced by "s"
+// and the va_list should only contain one argument, a full message with format substitutions
+// already filled.
+static void GULLOSLogWithType(os_log_t log, os_log_type_t type, char *s, ...) {
+  if (@available(iOS 9.0, macOS 10.11, *)) {
+    va_list args;
+    va_start(args, s);
+#if TARGET_OS_TV
```

Comment:
Should this be an LP_64 ifdef instead? and should it use va_list?  Please test on a 32 bit simulator. (iPhone 5).